conan/internal/api/new/oss_recipe.py

```diff
@@ -0,0 +1,100 @@
+config_yml = """\
+versions:
+  "{{version}}":
+    folder: all
+"""
+
+conandata_yml = """\
+sources:
+  "{{version}}":
+    url:
+      {% if url is defined -%}
+      - "{{url}}"
+      {% else -%}
+      - "http://put/here/the/url/to/release.1.2.3.zip"
+      {% endif %}
+    {% if sha256 is defined -%}
+    sha256: "{{sha256}}"
+    {%- else -%}
+    sha256: "Put here your tarball sha256"
+    {% endif -%}
+"""
+
+
+conanfile = """\
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, load, replace_in_file, save
+
+
+class {{package_name}}Recipe(ConanFile):
+    name = "{{name}}"
+    version = "{{version}}"
+    package_type = "library"
+
+    # Optional metadata
+    license = "<Put the package license here>"
+    author = "<Put your name here> <And your email here>"
+    url = "<Package recipe repository url here, for issues about the package>"
+    description = "<Description of {{ name }} package here>"
+    topics = ("<Put some tag here>", "<here>", "<and here>")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder)
```

Comment:
```suggestion
        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
```

Reply:
I had to remove that, because it was failing in the test if there is no root folder in the zip

Reply:
Maybe we could fake it in the test with something like:

```
        save_files(repo_folder, {"somefolder/CMakeLists.txt": cmake,
                                      "somefolder/pkg.h": gen_function_h(name="pkg"),
                                      "somefolder/pkg.cpp": gen_function_cpp(name="pkg")})
```

Because I think it will fail otherwise for the most real world zip sources file?

Reply:
Done